### ISSUE TITLE
oclint-json-compilation-database doesn't support `-p` option

### ISSUE THREAD

**Issue by AbigailBuccaneer** [oclint/oclint#435]
`oclint` supports specifying the path to `compile_commands.json` with `-p dir`. `oclint-json-compilation-database` doesn't understand this flag, meaning it's necessary to do `(cd dir && oclint-json-compilation-database)` which can be awkward for build systems.

Ideally it'd be possible to specify `-p` and it would both use that to find the file itself and pass it through to `oclint`.

**Comment by apmanol** [comment]
It does, but you need to follow the syntax rule of the script, for exampe:
```
/bin/oclint-json-compilation-database \
-- \
-o ${OCLINT_REPORT_PATH} -report-type ${OCLINT_REPORT_TYPE} \
-max-priority-1 ${OCLINT_MAX_PRIORITY_1} \
-max-priority-2 ${OCLINT_MAX_PRIORITY_2} \
-max-priority-3 ${OCLINT_MAX_PRIORITY_3} \
-p ${PWD} 

```

**Comment by AbigailBuccaneer** [comment]
It works if the path you want to pass is `$PWD`, but it doesn't work in general. `oclint-json-compilation-database` reads `./compile_commands.json` itself to form the `oclint` command line, and that's not configurable with `-p`. 

```
$ find . -name compile_commands.json
./build/compile_commands.json
$ oclint-json-compilation-database -- -p build/
Error: compile_commands.json not found at current location.
```

**Comment by ryuichis** [comment]
This is the reason we ask you to copy over `compile_commands.json` to the root of your project. And it's a `libClang` requirement.

**Comment by AbigailBuccaneer** [comment]
`libTooling` / `libClang`, and `oclint` itself, support this just fine. It's a requirement that the file is named `compile_commands.json`, but *not* that it's at the project root. The bug here is that `oclint-json-compilation-database` [hardcodes the location](https://github.com/oclint/oclint-json-compilation-database/blob/master/oclint-json-compilation-database#L16) that it reads `compile_commands.json` itself from:

```Python
CURRENT_WORKING_DIRECTORY = os.getcwd()
JSON_COMPILATION_DATABASE = CURRENT_WORKING_DIRECTORY + os.sep + "compile_commands.json"
```

https://github.com/oclint/oclint-json-compilation-database/pull/9 nearly fixed this, but not quite.
